rero_ils/modules/items/utils.py

```diff
@@ -112,6 +114,31 @@ def get_provisional_items_pids_candidate_to_delete():
         .filter('term', type=TypeOfItem.PROVISIONAL) \
         .filter('terms', status=[ItemStatus.ON_SHELF]) \
         .exclude('terms', pid=item_pids_with_fees)\
-        .source('pid')
+        .source(False)
+    for hit in query.scan():
+        yield Item.get_record(hit.meta.id)
+
+
+def update_late_expected_issue(dbcommit=False, reindex=False):
+    """Update status of issues with a passed `expected_date` to late status.
+
+    :param reindex: reindex record by record.
+    :param dbcommit: commit record to database.
+    :return: the number of updated issues.
+    """
+    from rero_ils.modules.items.api import Item, ItemsSearch
+
+    yesterday = datetime.now(timezone.utc) - timedelta(days=1)
+    yesterday = yesterday.strftime('%Y-%m-%d')
+    query = ItemsSearch() \
+        .filter('term', type=TypeOfItem.ISSUE) \
+        .filter('term', issue__status=ItemIssueStatus.EXPECTED) \
+        .filter('range', issue__expected_date={'lte': yesterday}) \
+        .source(False)
+    counter = 0
     for hit in query.scan():
```

Comment:
For count of things, `enumerate` is the fastest and preferred method. 
`count += 1` in loops should be avoided. But we have to initialise `count = 0`.

Reply:
corrected ;-)
Is it really quicker ? I'm interested to understand how it works.